pkg/dependencies/helmfactory.go

```diff
@@ -0,0 +1,138 @@
+package dependencies
+
+import (
+	"context"
+	"sync"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	configcli "github.com/aws/eks-anywhere/pkg/config"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/registrymirror"
+)
+
+// ExecutableBuilder builds the Helm executeble and returns it.
+type ExecutableBuilder interface {
+	BuildHelmExecutable(...executables.HelmOpt) *executables.Helm
```

Comment:
Okay, gotcha. I'll attempt to decouple this.

curious. Why do you think it's not ideal to depend on the executables package? This will make consumers have to depend on it, yes, but when are these dependencies acceptable in your opinion.

Reply:
in general, nowhere
`executables` is in general a implementation detail. It wraps around binaries to offer a go interface to access their functionality. Hence, no other package should depend on it, because no other package needs to know about the fact that the clients are implemented by wrapping binary executions.

Reply:
I've decoupled the `executables` dependency from this package by reversing the dependency and changing the `BuildHelmExecutable` method to implement an interface from the helm package:

```
func (b *ExecutablesBuilder) BuildHelmExecutable(opts ...HelmOpt) *Helm
```
to

```
func (b *ExecutablesBuilder) BuildHelmExecutable(opts ...helm.Opt) helm.ExecuteableClient
```

It was a little messy and had to tough other parts of the code to make the change